### ISSUE TITLE
cannot get fixes to run if added in RuleWalker.visitFunctionDeclaration custom rule

### ISSUE THREAD

**Issue by abrady0** [palantir/tslint#3749]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.5.2
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
import * from 'bar';
void foo(a: int, b: string) {
}
```

with `tslint.json` configuration:

```json
{
  /*
   * Possible values:
   * - the name of a built-in config
   * - a string that can be resolved as a module that exports the config object, e.g.:
   *   - an npm package name
   *   - an npm package name followed by the path to a file inside
   *   - a relative path to a JSON file
   */
  "extends": [
    "tslint-eslint-rules"
  ],
  "rules": {
    "move-functions": true,
    "brace-style": [
      true,
      "1tbs",
      { "allowSingleLine": true }
    ],
    "trailing-comma": [
      true,
      { "multiline": "always" }
    ],
    "curly": true,
    "eofline": true,
    "triple-equals": true,
    "ter-indent": [
      2,
      2,
      {
        "SwitchCase": 1
      }
    ],
    "no-eval": true,
    "no-invalid-this": true,
    "no-trailing-whitespace": true,
    "quotemark": [
      true,
      "single",
      "avoid-escape"
    ],
    "semicolon": [
      true,
      "always"
    ],
    "space-before-function-paren": [
      2,
      "never"
    ],
    "no-shadowed-variable": true,
    "no-constant-condition": 0,
    "no-regex-spaces": 0,
    "no-for-in-array": true,
    "no-floating-promises": true,
    "await-promise": [true, "PromiseLike"],
    /*
     * Any rules specified here will override those from the base config we are extending.
     */
    "no-consecutive-blank-lines": [ false ],
    "no-var-keyword": true,
    "prefer-const": false,
    "max-line-length": [ true, 150 ],
    "only-arrow-functions": [ false ],
    "forin": false,
    "no-var-requires": false,
    "object-literal-sort-keys": false,
    "object-literal-shorthand": false,
    "whitespace": [
      true,
      "check-branch",
      "check-decl",
      "check-operator",
      "check-module",
      "check-separator",
      "check-type",
      "check-typecast"
    ]
  },
  "rulesDirectory": [
    "scripts/tslint-rules/built"
  ]
}
```

#### Actual behavior

Running the below rule on the above file results in output:
```
Fixed 2 error(s) in foo.ts
```

and foo.ts having contents:
```ts

void foo(a: int, b: string) {
}
```

(that is, it says 2 errors were fixed but it appears that only one was)

Here is the custom declaration with the visitImportDeclaration that works, and the visitFunctionDeclaration that does not
```
export class Rule extends Lint.Rules.AbstractRule {
  public static FAILURE_STRING = 'fixing things up!!!';

  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
    return this.applyWithWalker(new FunctionWalker(sourceFile, this.getOptions()));
  }
}

class FunctionWalker extends Lint.RuleWalker {
  public visitImportDeclaration(node: ts.ImportDeclaration) {
    // create a fixer for this failure
    const fix = new Lint.Replacement(node.getStart(), node.getWidth(), "");

    // create a failure at the current position
    this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.FAILURE_STRING, fix));

    // call the base version of this visitor to actually parse this node
    super.visitImportDeclaration(node);
  }


  public visitFunctionDeclaration(node: ts.FunctionDeclaration) {
    // create a fixer for this failure
    const fix = new Lint.Replacement(node.getStart(), node.getWidth(), "");

    // create a failure at the current position
    this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.FAILURE_STRING, fix));

    // call the base version of this visitor to actually parse this node
    super.visitFunctionDeclaration(node);
  }
}

```
#### Expected behavior
I would expect the file to look as follows (i.e. have function and import removed):
```ts

```

**Comment by ajafff** [comment]
That's because your function is not a function. Functions in JavaScript (and TypeScript) start with the `function` keyword (not with the type as in your example).
Your example is actually a VoidExpression containing a call of `foo` with really weird arguments, followed by an empty block statement.

You should see a lot of syntax errors when compiling this code snippet.
